packages/payments-plugin/src/stripe/stripe.controller.ts

```diff
@@ -0,0 +1,125 @@
+import { BadRequestException, Controller, Headers, Post, Req } from '@nestjs/common';
+import {
+    ChannelService,
+    InternalServerError,
+    LanguageCode,
+    Logger,
+    Order,
+    OrderService,
+    OrderStateTransitionError,
+    PaymentMethod,
+    RequestContext,
+    TransactionalConnection,
+} from '@vendure/core';
+import Stripe from 'stripe';
+
+import { loggerCtx } from './constants';
+import { stripePaymentMethodHandler } from './stripe.handler';
+import { StripeService } from './stripe.service';
+import { IncomingMessageWithRawBody } from './types';
+
+@Controller('payments')
+export class StripeController {
+    constructor(
+        private connection: TransactionalConnection,
+        private channelService: ChannelService,
+        private orderService: OrderService,
+        private stripeService: StripeService,
+    ) {}
+
+    @Post('stripe')
+    async webhook(
+        @Headers('stripe-signature') signature: string | undefined,
+        @Req() request: IncomingMessageWithRawBody,
+    ): Promise<void> {
+        if (!signature) {
+            throw new BadRequestException('Missing stripe-signature header');
```

Comment:
The logs show the error message and the stack trace - I'm using the `DefaultLogger` below but the same info should be displayed by other custom loggers as well:

<img width="937" alt="image" src="https://user-images.githubusercontent.com/8788208/154177427-bbf78de5-2e16-4c12-9aba-ae724b4b44f1.png">

This is due to `ExceptionLoggerFilter`, [which automatically handles](https://github.com/vendure-ecommerce/vendure/blob/53babf654176d5d88cd7c1c7f88be98f8dd31a57/packages/core/src/api/middleware/exception-logger.filter.ts) these exceptions and log their data.

Compared to using `Logger.error()` and setting the status code in the response instead, we basically have the same output but no stack trace in the logs:

```typescript
@Post('stripe')
async webhook(
    @Headers('stripe-signature') signature: string | undefined,
    @Req() request: IncomingMessageWithRawBody,
    @Res() response: Response
): Promise<void> {
    if (!signature) {
        Logger.error('Missing stripe-signature header');
        response.status(HttpStatus.BAD_REQUEST).send('Missing stripe-signature header');
        return;
    }

    // ...
}
```

<img width="607" alt="image" src="https://user-images.githubusercontent.com/8788208/154176718-0fcedbab-7fac-4d5a-b72a-cee5aa21266b.png">

Should we stick to the former as it's more concise and generates more log data? If that's the case, I'll update the message to include the logger context:

```javascript
throw new BadRequestException(`[${loggerCtx}] Missing stripe-signature header`);
```

Reply:
You can pass a 3rd argument to `Logger.error()` for the stack trace like this:
```ts
Logger.error('Missing stripe-signature header', loggerCtx, e.stack);
```

In this case, we don't have an existing error object, but you could do this I think:
```ts
const message = 'Missing stripe-signature header'
Logger.error(message, loggerCtx, new Error(message).stack);
```

Worth trying out to see if it preserves the same level of information as just throwing. If so, it's preferable because it makes the logs more uniform, thus easier to parse.

Reply:
> it makes the logs more uniform, thus easier to parse

Gotcha. In fact, the stack trace is not important as all we need is the logger context and the message to see what went wrong. I'm updating it so the output is as follows:

<img width="592" alt="image" src="https://user-images.githubusercontent.com/8788208/154373361-c8f1da91-f8d3-4717-bd93-cd08b5ff00a8.png">